RPM improvements/fixes: lock down and fix %files section

jasonwbarnett opened the pull request:
- Fix what @glensc pointed out here: https://github.com/firehol/netdata/pull/734#discussion_r72586417
- Fix the logrotate configuration with what @glensc pointed out here: https://github.com/firehol/netdata/pull/734#discussion_r72586278
- Reorder `%files` section to make things more readable.
- Lock down everything under `/var` to be owned by `netdata` and to be locked down with `0700` permissions. This follows Redhat's best practices; can be seen in their httpd RPM.

jasonwbarnett commented:
@ktsaou @glensc take a look, but this is fairly minor changes/fixes.

ktsaou commented:
If it works for you, I'll merge it.
@glensc ?

glensc commented on netdata.spec.in:
```diff
@@ -129,27 +129,29 @@ fi
 %files
-%attr(-,netdata,netdata) %dir %{_localstatedir}/cache/%{name}
-%attr(-,netdata,netdata) %dir %{_localstatedir}/log/%{name}
+%defattr(-,root,root)
```
this is incomplete, use:

`%defattr(-,root,root,-)`

glensc commented on netdata.spec.in:
```diff
@@ -129,27 +129,29 @@ fi
+
 %dir %{_datadir}/%{name}
-%dir %{_sharedstatedir}/%{name}
+%attr(-,root,netdata) %{_datadir}/%{name}/web
```
this is not exactly the same as playing with `%defattr(644,root,netdata,755)`

with defattr one can force 644 for files and 755 for dirs. what you do is to assume permission bits what `make install` produced. this is probably fine, but double check that the files under web are permissions intended to.

glensc commented:
please also package registry %dir. i've pointed it out in some issue comments.

jasonwbarnett commented:
> this is not exactly the same as playing with %defattr(644,root,netdata,755)

Fixed

> please also package registry %dir. i've pointed it out in some issue comments.

Fixed

ktsaou commented:
let me know when you want it merged.

jasonwbarnett commented:
@ktsaou merge away!

ktsaou commented:
merged! thanks!

glensc commented on netdata.spec.in:
```diff
@@ -142,9 +142,10 @@ rm -rf $RPM_BUILD_ROOT
 
 %dir %{_datadir}/%{name}
-%attr(-,root,netdata) %{_datadir}/%{name}/web
+%attr(0644,root,netdata,0755) %{_datadir}/%{name}/web
```
err, what? have you actually tested this? %attr takes 3 arguments, it's defattr that has 4 arguments
